sdcm/docker.py

```diff
@@ -215,33 +239,150 @@ def destroy(self):
             node.destroy(force=True)
 
 
-class ScyllaDockerCluster(DockerCluster, cluster.BaseScyllaCluster):  # pylint: disable=abstract-method
+class ScyllaDockerCluster(cluster.BaseScyllaCluster, DockerCluster):  # pylint: disable=abstract-method
 
     def __init__(self, **kwargs):
-        self._user_prefix = kwargs.get('user_prefix', cluster.DEFAULT_USER_PREFIX)
-        self._node_prefix = '%s-%s' % (self._user_prefix, BASE_NAME)
-        super(ScyllaDockerCluster, self).__init__(node_prefix=kwargs.get('node_prefix', self._node_prefix),
+        user_prefix = kwargs.get('user_prefix')
+        cluster_prefix = cluster.prepend_user_prefix(user_prefix, 'db-cluster')
+        node_prefix = cluster.prepend_user_prefix(user_prefix, 'db-node')
+
+        self.aws_extra_network_interface = False  # TODO: move to Base
+        super(ScyllaDockerCluster, self).__init__(node_prefix=node_prefix,
+                                                  cluster_prefix=cluster_prefix,
                                                   **kwargs)
 
-    @retrying(n=30, sleep_time=3, allowed_exceptions=(cluster.ClusterNodesNotReady, DockerCommandError))
+    def node_setup(self, node, verbose=False, timeout=3600):
+        endpoint_snitch = self.params.get('endpoint_snitch')
+        seed_address = ','.join(self.seed_nodes_ips)
+
+        node.wait_ssh_up(verbose=verbose)
+        # node.install_scylla_debuginfo()
+
+        self.node_config_setup(node, seed_address, endpoint_snitch)
+
+        node.stop_scylla_server(verify_down=False)
+        # clear data folder to drop wrong cluster name data
+        node.remoter.run('sudo rm -Rf /var/lib/scylla/data/*')
+        node.start_scylla_server(verify_up=False)
+
+        node.wait_db_up(verbose=verbose, timeout=timeout)
+        node.check_nodes_status()
+        self.clean_replacement_node_ip(node)
+
+    @cluster.wait_for_init_wrap
     def wait_for_init(self, node_list=None, verbose=False, timeout=None):   # pylint: disable=unused-argument,arguments-differ
         node_list = node_list if node_list else self.nodes
         for node in node_list:
             node.wait_for_status_running()
-        return self.check_nodes_up_and_normal(node_list)
+        self.wait_for_nodes_up_and_normal(node_list)
+
+    def get_scylla_args(self):
```

Comment:
thinks if to remove much more here, since crashing with the default bad alloc flags

Reply:
check again, seem like now need